src/source/distortion.rs

```diff
@@ -0,0 +1,116 @@
+use std::time::Duration;
+
+use super::SeekError;
+use crate::common::{ChannelCount, SampleRate};
+use crate::Source;
+
+/// Internal function that builds a `Distortion` object.
+pub fn distortion<I>(input: I, gain: f32, threshold: f32) -> Distortion<I>
```

Comment:
we have decided to stop making these functions public, that still needs to be done to the rest of the source code. So if you want we can start with distortion